### DISCUSSION TITLE
Unable to Install WSL in Windows Terminal, Stuck at 0%

### DISCUSSION THREAD

**Discussion by Ayomiposi233** [Cyfrin/foundry-full-course-cu#3017] [Q&A] [answered]
Hello Guys, 
Can someone give me any advice that can help concerning WSL install? It is just stuck at 0% in my windows terminal. My pc is a Windows 11pro, any help will be appreciated, Thanks

**Comment by EngrPips** [comment, marked as answer]
Hello @Ayomiposi233, I asked Gemini AI, and the below are the troubleshooting suggestions

**Troubleshooting WSL Installation Stuck at 0%**

Here are some steps you can take to resolve the WSL installation issue:

**1. Enable Virtual Machine Platform:**

* **Check if it's enabled:**
  - Open **Settings** > **Apps** > **Optional Features**.
  - Search for "Virtual Machine Platform" and ensure it's enabled. If not, click **Install**.
* **Restart your PC:** After enabling the feature, restart your computer.

**2. Update Windows:**

* Ensure your Windows 11 is up-to-date. Install any pending updates.

**3. Run WSL Installation as Administrator:**

* **Open PowerShell as Administrator:** Right-click on the PowerShell icon and select "Run as administrator".
* **Run the installation command:** Type `wsl --install -d Ubuntu` and press Enter.

**4. Try a Different Distribution:**

* If you're installing a specific distribution, try installing a different one like Ubuntu or Debian. You can change the distribution later.

**5. Check Network Connectivity:**

* Ensure you have a stable internet connection. A slow or unstable connection can cause installation issues.

**6. Disable Antivirus Temporarily:**

* Sometimes, antivirus software can interfere with the installation process. Temporarily disable your antivirus and try the installation again.

**7. Manual Installation:**

* If the above methods don't work, you can try manually installing WSL:
  1. Download the desired Linux distribution (e.g., Ubuntu) from the Microsoft Store.
  2. Install it like any other app.

**Additional Tips:**

* **Clear the WSL Cache:** If you've tried previous installations, clearing the WSL cache might help.
* **Check for Errors in Event Viewer:** Look for any errors related to WSL in the Windows Event Viewer.
* **Consult Microsoft Documentation:** Refer to the official Microsoft documentation for the latest troubleshooting steps.

If you're still facing issues after trying these solutions, please provide more details about the specific error messages or any unusual behavior you're observing. This will help in pinpointing the exact problem and finding a suitable solution.

**Reply by EngrPips** [reply]
Please try these troubleshooting suggestions and let us know if it helps at all.

**Reply by Ayomiposi233** [reply]
The number 1 tip on the list worked well. I have successfully installed WSL and Ubuntu. Thank you so much.

**Reply by EngrPips** [reply]
Awesome, I am glad that helped, Keep crushing it.

**Comment by bouchaib87** [comment]
https://linuxconfig.org/ubuntu-22-04-on-wsl-windows-subsystem-for-linux

**Reply by EngrPips** [reply]
Thanks for the heads-up.

**Comment by dinshows15** [comment]
Guys, I used following link 
https://github.com/microsoft/WSL/releases/download/2.6.0/wsl.2.6.0.0.x64.msi

And it worked for me to install WSL2

**Reply by EngrPips** [reply]
Thanks for the Heads-up.
